tracecontext.go

```diff
@@ -169,9 +169,40 @@ type TraceState struct {
 
 // NewTraceState returns a TraceState based on entries.
 func NewTraceState(entries ...TraceStateEntry) TraceState {
-	out := TraceState{}
+	var out TraceState
 	var last *TraceStateEntry
+	var recorded uint8
+	// Range over the entries and find duplicate `es` vendor keys. When 'es'
```

Comment:
I'm finding this method quite difficult to follow now. Would this alternative work?

```go
var haveElastic bool
var last *TraceState
for _, e := range entries {
	if e.Key == elasticTracestateVendorKey {
		if haveElastic {
			// Discard duplicate "es" entries; keep the last entry's value.
			out.head.value = e.value
			continue
		}
		haveElastic = true
		e.next = last
		last = nil // cause elastic to be moved to head
	}
	e := e // copy
	if last == nil {
		out.head = &e
	} else {
		last.next = &e
	}
	last = &e
}
if haveElastic {
	out.parseElasticTracestateError = out.parseElasticTracestate(out.head)
}
```

Reply:
BTW we didn't previously move the elastic entry to the head because IIANM that's only expected when _mutating_ tracestate.

Reply:
I mostly kept the suggestion intact, but had to make some changes in 9270b6a to not lose any entries that were found previous to the `es` key.